pkg/morph/event/reputation/put_notary.go

```diff
@@ -0,0 +1,86 @@
+package reputation
+
+import (
+	"fmt"
+
+	"github.com/nspcc-dev/neofs-node/pkg/morph/event"
+)
+
+func (p *Put) setEpoch(v uint64) {
+	p.epoch = v
+}
+
+func (p *Put) setPeerID(v []byte) error {
+	if v == nil {
+		return nil
+	}
+
+	if ln := len(v); ln != peerIDLength {
+		return fmt.Errorf("peer ID is %d byte long, expected %d", ln, peerIDLength)
+	}
+
+	var publicKey [33]byte
+	copy(publicKey[:], v)
+	p.peerID.SetPublicKey(publicKey)
+
+	return nil
+}
+
+func (p *Put) setValue(v []byte) error {
+	if v == nil {
+		return nil
+	}
+
+	return p.value.Unmarshal(v)
+}
+
+var fieldSetters = []func(*Put, []byte) error{
+	// order on stack is reversed
+	(*Put).setValue,
+	(*Put).setPeerID,
+}
+
+const (
+	// PutNotaryEvent is method name for reputation put operations
+	// in `Reputation` contract. Is used as identifier for notary
+	// put reputation requests.
+	PutNotaryEvent = "put"
+)
+
+var errUnexpectedArgumentAmount = fmt.Errorf("unexpected arguments amount in %s call", PutNotaryEvent)
```

Comment:
suggest using https://github.com/nspcc-dev/neofs-node/blob/master/pkg/morph/event/notary.go#L48 to be sure that error messages have the same form

that approach is used in other notary parsers